src/moments.jl

```diff
@@ -158,6 +158,39 @@ function mean_and_std(x::RealArray, w::AbstractWeights, dims::Int;
 end
 
 
+"""
+    n_mean_var(x; mean=nothing, corrected=true) -> (n, mean, var)
+
+Return the length, mean, and variance of `x`.
+"""
+function n_mean_var(x; mean=nothing, corrected=true)
```

Comment:
What's the motivation for adding this function? Everything is already implemented, isn't it?

Reply:
> What's the motivation for adding this function? Everything is already implemented, isn't it?

Computes everything in a single pass, and does the work for general iterators rather than just arrays.

Reply:
`mean` and `var` already work for general iterators. Additionally, code such as `mapreduce(.+, x)` is probably less efficient than existing implementations. I suggest removing this method again.

Reply:
IIUC the goal is to also compute the number of elements on the fly, right? It's unfortunate that the implementation for `var` lives in Statistics, otherwise we would have adapted it to optionally compute the number of elements rather than duplicating it.

Reply:
> `mean` and `var` already work for general iterators. Additionally, code such as `mapreduce(.+, x)` is probably less efficient than existing implementations. I suggest removing this method again.

`mean` and `var` already work for general iterators, but `length` doesn't. The implementation with `length` only works for vectors and similar functions with an already-defined `length` function.

Reply:
This is almost identical to the code for `sem` that was already in use, with the exception that `mean` can be passed as an argument.